src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs

```diff
@@ -249,9 +249,13 @@ public sealed class RInteractiveEvaluator : IInteractiveEvaluator {
         }
 
         private void WriteError(string message) {
-            if (CurrentWindow != null) {
-                _coreShell.DispatchOnUIThread(() => CurrentWindow?.WriteError(message));
-            }
+            _coreShell.DispatchOnUIThread(() => {
```

Comment:
Let's do double checking here, to avoid expensive dispatches when there's nothing to do.